library/std/src/sys/pal/hermit/net.rs

```diff
@@ -225,17 +225,11 @@ impl Socket {
     }
 
     pub fn write_vectored(&self, bufs: &[IoSlice<'_>]) -> io::Result<usize> {
-        let mut size: isize = 0;
-
-        for i in bufs.iter() {
-            size += cvt(unsafe { netc::write(self.0.as_raw_fd(), i.as_ptr(), i.len()) })?;
-        }
-
-        Ok(size.try_into().unwrap())
+        crate::io::default_write_vectored(|b| self.write(b), bufs)
```

Comment:
huh, this appears to be the exact same implementation as what's on the left. weird.

Reply:
Yes, but is_write_vectored was wrong.

Reply:
yep!

Reply:
You could probably just delete the implementation altogether, it'll fall back to the default automatically.

Reply:
This is the impl of Socket, tho, innit?

Or do you mean that the `write_vectored` impls will resolve to the trait and then dispatch to that one's default impl?

Reply:
> This is the impl of Socket, tho, innit?

Oh yeah. Nevermind my comment then, didn't notice.